include/freerdp/transport_io.h

```diff
@@ -42,20 +59,26 @@ extern "C"
 	typedef BOOL (*pTransportGetPublicKey)(rdpTransport* transport, const BYTE** data,
 	                                       DWORD* length);
 	typedef BOOL (*pTransportSetBlockingMode)(rdpTransport* transport, BOOL blocking);
+	typedef rdpTransportLayer* (*pTransportConnectLayer)(rdpTransport* transport,
+	                                                     const char* hostname, int port,
+	                                                     DWORD timeout);
+	typedef BOOL (*pTransportAttachLayer)(rdpTransport* transport, rdpTransportLayer* layer);
 
 	struct rdp_transport_io
 	{
-		pTCPConnect TCPConnect;
-		pTransportFkt TLSConnect;
-		pTransportFkt TLSAccept;
-		pTransportAttach TransportAttach;
-		pTransportFkt TransportDisconnect;
-		pTransportRWFkt ReadPdu;  /* Reads a whole PDU from the transport */
-		pTransportRWFkt WritePdu; /* Writes a whole PDU to the transport */
-		pTransportRead ReadBytes; /* Reads up to a requested amount of bytes from the transport */
-		pTransportGetPublicKey GetPublicKey;
-		pTransportSetBlockingMode SetBlockingMode;
-		UINT64 reserved[54]; /* Reserve some space for ABI compatibility */
+		ALIGN64 pTCPConnect TCPConnect;
```

Comment:
while most likely does not change much, this alignment might break `ABI/API` for existing 3.x

Reply:
Without the alignment it is probably already going to break on 32-bit system whenever adding new entries. I think it is better to fix it now.